raft/server.cc

```diff
@@ -1142,7 +1143,14 @@ future<> server_impl::read_barrier(seastar::abort_source* as) {
             leader = _fsm->current_leader();
         } else {
             auto applied = _applied_idx;
-            auto res = co_await get_read_idx(leader, as);
+            read_barrier_reply res;
+            try {
+                res = co_await get_read_idx(leader, as);
+            } catch (const seastar::rpc::closed_error& e) {
+                logger.trace("[{}] read_barrier on {} resulted in seastar::rpc::closed_error; retrying", _id, leader);
+                leader = server_id{};
```

Comment:
We may enter a busy loop here until leader changes. We need to sleep, but we raft code to be clock neutral.